tf/edgeml/graph/protoNN.py

```diff
@@ -0,0 +1,141 @@
+'''
+ProtoNN Graph construction
+'''
+import numpy as np
+import tensorflow as tf
+
+
+class ProtoNN:
+    def __init__(self, inputDimension, projectionDimension, numPrototypes,
+                 numOutputLabels, gamma,
+                 W = None, B = None, Z = None):
+        '''
+        inputDimension: input data dimension. A [-1, inputDimension] matrix
+            is expected as part of __call__ method.
+        projectionDimension: (model paramter)
+        numPrototypes: (model parameter)
+        numOutputLabels: The number of output labels or classes
+        W, B, Z: initial value to use for W, B, Z
+        Expected Dimensions:
+        W   d x d_cap
+        B   d_cap x m
+        Z   L x m
+        X   n x d
+        '''
```

Comment:
Change the d, d_cap to the the elongated variable names like inputDimension etc.,